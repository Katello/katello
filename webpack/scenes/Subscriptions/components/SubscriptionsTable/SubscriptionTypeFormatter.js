import React from 'react';
import { Link } from 'react-router-dom';
import { urlBuilder } from 'foremanReact/common/urlHelpers';

export const subscriptionTypeFormatter = (value, { rowData }) => {
  let cellContent;

  if (rowData.virt_only === false) {
    cellContent = __('Physical');
  } else if (rowData.hypervisor) {
    cellContent = (
      <span>
        {__('Guests of')}
        {' '}
        <Link to={urlBuilder('content_hosts', '', rowData.hypervisor.id)}>{rowData.hypervisor.name}</Link>
      </span>
    );
  } else if (rowData.unmapped_guest) {
    cellContent = __('Temporary');
  } else {
    cellContent = __('Virtual');
  }

  return (
    <td>
      {cellContent}
    </td>
  );
};

export default subscriptionTypeFormatter;
