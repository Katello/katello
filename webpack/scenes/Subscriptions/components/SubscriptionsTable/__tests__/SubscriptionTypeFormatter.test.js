import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import { subscriptionTypeFormatter } from '../SubscriptionTypeFormatter';

describe('subscriptionTypeFormatter', () => {
  const data = rowData => ({
    rowData,
  });

  it('renders physical subscriptions', async () => {
    const formatter = subscriptionTypeFormatter(null, data({ virt_only: false }));

    expect(toJson(shallow(formatter))).toMatchSnapshot();
  });

  it('renders temporary subscriptions', async () => {
    const formatter = subscriptionTypeFormatter(null, data({ unmapped_guest: true }));

    expect(toJson(shallow(formatter))).toMatchSnapshot();
  });

  it('renders virtual subscriptions', async () => {
    const formatter = subscriptionTypeFormatter(null, data({}));

    expect(toJson(shallow(formatter))).toMatchSnapshot();
  });

  it('renders link to a host', async () => {
    const formatter = subscriptionTypeFormatter(null, data({ hypervisor: { name: 'host.example.com', id: 83 } }));

    expect(toJson(shallow(formatter))).toMatchSnapshot();
  });
});
