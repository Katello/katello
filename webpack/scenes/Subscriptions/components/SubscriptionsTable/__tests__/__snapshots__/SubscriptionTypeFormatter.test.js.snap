// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`subscriptionTypeFormatter renders link to a host 1`] = `
<td>
  <span>
    Guests of
     
    <Link
      to="/content_hosts/83/"
    >
      host.example.com
    </Link>
  </span>
</td>
`;

exports[`subscriptionTypeFormatter renders physical subscriptions 1`] = `
<td>
  Physical
</td>
`;

exports[`subscriptionTypeFormatter renders temporary subscriptions 1`] = `
<td>
  Temporary
</td>
`;

exports[`subscriptionTypeFormatter renders virtual subscriptions 1`] = `
<td>
  Virtual
</td>
`;
